lib/bin/lwm2m_carrier/os/lwm2m_os.c

```diff
@@ -162,9 +162,8 @@ static void work_handler(struct k_work *work)
 
 		timeout = get_timeout_value(timeout, lwm2m_work);
 
-		/* FIXME: handle error return from k_delayed_work_submit(). */
-		(void)k_delayed_work_submit(&lwm2m_work->work_item,
-					    K_MSEC(timeout));
+		/* FIXME: handle error return from k_work_reschedule(). */
```

Comment:
No need anymore, reschedule is always successful in some way or another :) This could additionally be a `schedule` now, as it's scheduling itself.

Reply:
So when a task reschedules itself, is it preferred to use `_schedule` instead of `_reschedule`?
I would guess the behavior is the same.

Reply:
`schedule`, but the outcome is the same